### ISSUE TITLE
Add ANSI syntax highlighting to overload matching errors

### ISSUE THREAD

**Issue by postmodern** [crystal-lang/crystal#11755]
## Feature Request

In order to improve readability of the overload matching error, the list of overloaded method signatures could be syntax highlighted with ANSI color codes. This would improve the readability of rather verbose method signatures.

```
Overloads are:
 - Test#test(arg1 : Int32, arg2 : Int32, arg3 : UInt8)
 - Test#test(arg1 : Int, arg2 : Int32, arg3 : Int::Unsigned)
 - Test#test(arg1 : UInt32, arg2 : UInt32, arg3 : UInt32, &block : (UInt32, UInt32, UInt32 -> Int64))
```

Elements we could highlight:

* `,`
* `:`
* `(` `)`
* `->`
* `#`
* argument name
* Type names

I will let others decide what color scheme should be used, or whether ANSI bold or bright colors should be used to highlight specific elements in the method signature.

**Comment by wyhaines** [comment]
FWIW, I have a PR that I am playing with to do both this and syntax highlighting of the code lines in a backtrace.

Currently:

![image](https://user-images.githubusercontent.com/12992/150293165-7a739c32-79c6-4074-ae2a-eeb520c8b65c.png)

![image](https://user-images.githubusercontent.com/12992/150293209-ac78df12-1f89-4d40-a763-ca35b4e7630f.png)

**Comment by postmodern** [comment]
Would also be nice to syntax highlight type unions (`(Int32 | String | UInt32)`). Highlighting the `(`, `)`, and `|` chars in a different color than the type names would help visually break it up.

**Comment by wyhaines** [comment]
I agree and I will make those additions tomorrow, after I sleep, and then test it on some real code, and then I will drop the images into this thread.

**Comment by HertzDevil** [comment]
Duplicate of #11106 (see also https://github.com/crystal-lang/crystal/issues/11106#issuecomment-902261083)

**Comment by postmodern** [comment]
#11106 seems to be about indicating the compatible arguments to find the optimal overload. This feature request is just to add basic syntax highlighting to make it easier to read method signatures and type signatures in error output.

**Comment by vlazar** [comment]
Also good to keep this comment in mind https://github.com/crystal-lang/crystal/issues/11106#issuecomment-902261083

**Comment by straight-shoota** [comment]
Maybe we should really just use `Crystal::SyntaxHighlighter::Colorize` for all code that is printed by the compiler, including method signatures and type unions.

**Comment by wyhaines** [comment]
This is essentially what the PR that I am working on does.


Kirk Haines
